tests/e2e/specs/backend/product-query/product-filters.test.ts

```diff
@@ -0,0 +1,273 @@
+ *
+ * There are also some labels that are used repeatedly, but we don't scope them
+ * in favor of readability. Unlike selectors, those label are visible to end
+ * users, so it's easier to understand what's going on if we don't scope them.
+ * Those labels can get upated in the future, but the tests will fail and we'll
+ * know to update them.
+ */
+const SELECTORS = {
+	productFiltersDropdownButton: {
+		default:
+			'.components-tools-panel-header .components-dropdown-menu button[aria-expanded="false"]',
+		expanded:
+			'.components-tools-panel-header .components-dropdown-menu button[aria-expanded="true"]',
+	},
+	productFiltersDropdown:
+		'.components-dropdown-menu__menu[aria-label="Product filters options"]',
+	productFiltersDropdownItem: '.components-menu-item__button',
+	editorPreview: {
+		productsGrid: 'ul.wp-block-post-template',
+		productsGridItem:
+			'ul.wp-block-post-template > li.block-editor-block-preview__live-content',
+	},
+	productsGrid: `${ block.class } ul.wp-block-post-template`,
+	productsGridItem: `${ block.class } ul.wp-block-post-template > li.product`,
+	formTokenFieldLabel: '.components-form-token-field__label',
+	tokenRemoveButton: '.components-form-token-field__remove-token',
+};
+
+const toggleProductFilter = async ( filterName: string ) => {
+	const productFiltersPanel = await findToolsPanelWithTitle(
+		'Product filters'
+	);
+	await expect( productFiltersPanel ).toClick(
+		SELECTORS.productFiltersDropdownButton.default
+	);
+	await canvas().waitForSelector( SELECTORS.productFiltersDropdown );
+	await expect( canvas() ).toClick( SELECTORS.productFiltersDropdownItem, {
+		text: filterName,
+	} );
+	await expect( productFiltersPanel ).toClick(
+		SELECTORS.productFiltersDropdownButton.expanded
+	);
+};
+
+const resetProductQueryBlockPage = async () => {
+	await visitBlockPage( `${ block.name } Block` );
+	await waitForCanvas();
+	await setPostContent( '' );
+	await insertBlock( block.name );
+	await saveOrPublish();
+};
+
+const getPreviewProducts = async (): Promise< ElementHandle[] > => {
+	await canvas().waitForSelector( SELECTORS.editorPreview.productsGrid );
+	return await canvas().$$( SELECTORS.editorPreview.productsGridItem );
+};
+
+const getFrontEndProducts = async (): Promise< ElementHandle[] > => {
+	await canvas().waitForSelector( SELECTORS.productsGrid );
+	return await canvas().$$( SELECTORS.productsGridItem );
+};
+
+describeOrSkip( GUTENBERG_EDITOR_CONTEXT === 'gutenberg' )(
+	'Product Query > Products Filters',
+	() => {
+		let productFiltersPanel: ElementHandle< Node >;
+		beforeEach( async () => {
+			/**
+			 * Reset the block page before each test to ensure the block is
+			 * inserted in a known state. This is also needed to ensure each
+			 * test can be run individually.
+			 */
+			await resetProductQueryBlockPage();
+			await openBlockEditorSettings( { isFSEEditor: false } );
+			await selectBlockByName( block.slug );
+			productFiltersPanel = await findToolsPanelWithTitle(
```

Comment:
This is another suggestion, so feel free to ignore, but again coming from experience in such automations. Usually I found that prefixing variables containing elements/nodes with the `$` makes them more recognizeable at a glance. This patter is used also by Chrome devtools, when you use the console and do something like `$0`. After jQuery, in the JS world, I feel that `$` has become associated with DOM nodes. So that's a nice way to make things more scannable.

Reply:
Prefixing element variables with `$` is my habit from the jQuery age as well, so yes, let's do that.